src/redux/movies.ts

```diff
@@ -0,0 +1,121 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { Movie } from 'types/movie';
+import { RootState } from './store';
+
+interface MoviesState {
+  movies: Movie[];
+  selectedMovie?: Movie;
+  moviesCount: number;
+}
+interface fetchProps {
+  searchText?: string;
+  searchBy?: string;
+  sortBy?: string;
+  sortOrder?: string;
+  filter?: string[];
+}
+const initialState: MoviesState = {
+  movies: [],
+  selectedMovie: undefined,
+  moviesCount: 0,
+};
+export const fetchMoviesAction = createAsyncThunk(
+  'moviesFetch',
+  (params: fetchProps) => {
+    return fetchMovies(params);
+  },
+);
+
+const fetchMovies = async ({
+  searchText,
+  searchBy,
+  sortBy,
+  sortOrder = 'asc',
+  filter,
+}: fetchProps) => {
+  const queryParams = [
+    searchText ? `search=${searchText}` : '',
+    searchBy ? `&searchBy=${searchBy}` : '',
+    sortBy ? `&sortBy=${sortBy}` : '',
+    filter?.length ? `&filter=${filter.join()}` : '',
+    `sortOrder=${sortOrder}`,
+  ]
+    .filter(Boolean)
+    .join('&');
+  const response = await fetch(`http://localhost:4000/movies?${queryParams}`);
+
+  return await response.json();
+};
+
+export const addMovieAction = createAsyncThunk(
+  'addMovie',
+  async (movie: Movie) => {
+    await fetch('http://localhost:4000/movies', {
+      method: 'POST',
+      body: JSON.stringify(movie),
+      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
```

Comment:
I see that configuration of your HTTP client repeats several times. Let's try to avoid copy-past and use some base HTTP client setup. Take a look at my example [here](https://github.com/dotnet-fizzyy/Diploma/blob/V2/web/src/api/baseApi.ts#L3). I was using `axios` package, but there are no issues with migration to default `fetch`